### PR TITLE
feat(ecs): support for explicit activation of the circuit breaker

### LINKED ISSUE #27131: (ecs): Deployment Circuit Breaker cannot be turned off via CDK
### Describe the bug

I believe this is similar https://github.com/aws/aws-cdk/pull/25840, but for the `DeploymentCircuitBreaker`.

We are unable to turn off the DeploymentCircuitBreaker from an ECS service via CDK.  

### Expected Behavior

Deploying an ECS service with an unset `.circuitBreaker` will turn off the ECS Deployment Circuit Breaker if it was previously turned on.

### Current Behavior

When setting the `.circuitBreaker` property on an ECS service in CDK, we see the following added to the resulting ECS::Service CFN:
```
   "Type": "AWS::ECS::Service",
   "Properties": {
    "DeploymentConfiguration": {
     "DeploymentCircuitBreaker": {
      "Enable" : true,
      "Rollback" : false,
     },
    ...
    },
   ...
```

When attempting to remove the circuit breaker by not setting the `.circuitBreaker` property on the CDK service, the `DeploymentCircuitBreaker` is no longer set in the resulting CFN.  When this new stack (with a removed `DeploymentCircuitBreaker`), ECS does not turn off the circuit breaker.  I believe that this is due to ECS not treating missing values as "different" when processing the CFN.

Manually editing the CFN to explicitly set the `"Enable": false` in the CFN does cause ECS to turn off the circuit breaker, e.g.:
```
   "Type": "AWS::ECS::Service",
   "Properties": {
    "DeploymentConfiguration": {
     "DeploymentCircuitBreaker": {
      "Enable" : false,
      "Rollback" : false,
     },
    ...
    },
   ...
```

Unfortunately, this ability to explicitly disable the circuit breaker is not provided by the L2 CDK construct, and instead it is implied that not settting `.circuitBreaker` on the service will achieve this.

### Reproduction Steps

See "Current Behavior"

### Possible Solution

I believe the right way to fix this is explicitly disable the DeploymentCircuitBreaker in the CFN when the `.circuitBreaker` property is unset.

### Additional Information/Context

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ecs.FargateService.html#circuitbreaker

### CDK CLI Version

2.93.0

### Framework Version

_No response_

### Node.js Version

14.21.3

### OS

Linux

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by badmintoncryer**
This PR has enabled explicit disabling of the circuit breaker. 

```ts
declare const cluster: ecs.Cluster;
declare const taskDefinition: ecs.TaskDefinition;
const service = new ecs.FargateService(this, 'Service', {
  cluster,
  taskDefinition,
  circuitBreaker: {
    enable: true, // added
    rollback: true
  },
});
```

This is useful for removing a circuit breaker that has been set previously.

Closes #27131.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts [hunk 1]
```diff
@@ -68,6 +68,12 @@ export interface DeploymentController {
  * The deployment circuit breaker to use for the service
  */
 export interface DeploymentCircuitBreaker {
+  /**
+  * Whether to enable the deployment circuit breaker logic
+  * @default true
+  */
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts:74]:
```suggestion
  /**
   * Whether to enable the deployment circuit breaker logic
   * @default - true
   */
```

File: packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts [hunk 2]
```diff
@@ -68,6 +68,12 @@ export interface DeploymentController {
...
+  readonly enable?: boolean;
+
   /**
    * Whether to enable rollback on deployment failure
    * @default false
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts:79]:
```suggestion
   *
   * @default - false
```

File: packages/aws-cdk-lib/aws-ecs/README.md [hunk 3]
```diff
@@ -692,18 +692,20 @@ new ecs.ExternalService(this, 'Service', {
...
-deployment circuit breaker and optionally enable `rollback` for automatic rollback. See [Using the deployment circuit breaker](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-type-ecs.html)
-for more details.
+The Amazon ECS [deployment circuit breaker](https://aws.amazon.com/tw/blogs/containers/announcing-amazon-ecs-deployment-circuit-breaker/) automatically rolls back unhealthy service deployments, eliminating the need for manual intervention. By including the `circuitBreaker` parameter, you can activate this feature. Within `circuitBreaker`, the `enable` parameter is optional and defaults to true, which turns on the circuit breaker. If `circuitBreaker` is not specified, the feature remains disabled. You also have the option to explicitly set `enable` to false, which is useful for disabling an active circuit breaker. Additionally, specifying `rollback` triggers automatic rollback of deployments.
+
+See [Using the deployment circuit breaker](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-type-ecs.html) for more details.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ecs/README.md:697]:
```suggestion
The Amazon ECS [deployment circuit breaker](https://aws.amazon.com/tw/blogs/containers/announcing-amazon-ecs-deployment-circuit-breaker/)
automatically rolls back unhealthy service deployments, eliminating the need for manual intervention.

Use `circuitBreaker` to enable the deployment circuit breaker which determines whether a service deployment
will fail if the service can't reach a steady state.
You can optionally enable `rollback` for automatic rollback.
You can disable the circuit breaker by omitting the `circuitBreaker` property or setting `enabled` to `false`.

See [Using the deployment circuit breaker](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-type-ecs.html) for more details.
```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Looks good overall 👍
Just some minor documentation adjustments.

**Comment by badmintoncryer** [comment]
@lpizzinidev Thank you for your review and suggestion. I updated documentation.

**Review by lpizzinidev — APPROVED** [review]

File: packages/aws-cdk-lib/aws-ecs/README.md [hunk 4]
```diff
@@ -692,18 +692,26 @@ new ecs.ExternalService(this, 'Service', {
 
 ### Deployment circuit breaker and rollback
 
-Amazon ECS [deployment circuit breaker](https://aws.amazon.com/tw/blogs/containers/announcing-amazon-ecs-deployment-circuit-breaker/)
-automatically rolls back unhealthy service deployments without the need for manual intervention. Use `circuitBreaker` to enable
-deployment circuit breaker and optionally enable `rollback` for automatic rollback. See [Using the deployment circuit breaker](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-type-ecs.html)
-for more details.
+The Amazon ECS [deployment circuit breaker](https://aws.amazon.com/tw/blogs/containers/announcing-amazon-ecs-deployment-circuit-breaker/)
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-ecs/README.md:695]:
```suggestion
Amazon ECS [deployment circuit breaker](https://aws.amazon.com/tw/blogs/containers/announcing-amazon-ecs-deployment-circuit-breaker/)
```

File: packages/aws-cdk-lib/aws-ecs/README.md [hunk 5]
```diff
@@ -692,18 +692,26 @@ new ecs.ExternalService(this, 'Service', {
...
+automatically rolls back unhealthy service deployments, eliminating the need for manual intervention.
+
+Use `circuitBreaker` to enable the deployment circuit breaker which determines whether a service deployment
+will fail if the service can't reach a steady state.
+You can optionally enable `rollback` for automatic rollback.
+You can disable the circuit breaker by omitting the `circuitBreaker` property or setting `enabled` to `false`.
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-ecs/README.md:701]: Is this accurate that omitting the `circuitBreaker` property will disable the circuit breaker? I believe that was the original issue and I don't think this PR that has directly altered that behaviour.

**Review by paulhcsun — COMMENTED** [review]
Overall looks good, just one question about the README.

File: packages/aws-cdk-lib/aws-ecs/README.md (hunk 5 above)
Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-ecs/README.md]: I think you're right. It appeared to be an unnecessary detail, so I removed it.

**Review by paulhcsun — APPROVED** [review]
Nice work @badmintoncryer!

**Pull request merged**